blockchain/blockchain.go

```diff
@@ -902,6 +902,15 @@ func (b *Blockchain) WriteFullBlock(fblock *types.FullBlock, source string) erro
 		return err
 	}
 
+	if fblock.Receipts == nil { // Fetch the block receipts from cache if not already set
+		blockReceipts, receiptsErr := b.extractBlockReceipts(block)
```

Comment:
I do not want to do this. The last performance bug was due to not being able to keep track of when was `extractBlockReceipts` being used, which is why `FullBlock` and this method were introduced.

The `extractBlockReceipts` method will eventually get deprecated once `ibft` is removed.

Also, the initial logic of the function changes a lot, from inserting only into the storage the block data entries to maybe also compute the full block. When is this necessary?

Reply:
I didn't want to change the `blockchain.go` a lot just because of the `syncer`, so that function you added looked as the easiest place to change, also with the smallest amount of changes as well.

I can do it another way, but it will require changing `blockchain.go` other functions.